js/src/useFlow.types.ts

```diff
@@ -4,14 +4,14 @@ export type Next<T> = (currentValue: string | null) => Promise<string>;
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export type Next1<T, T1> = (
   currentValue: string | null,
-  arg1: T1,
+  arg1: any,
 ) => Promise<string>;
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export type Next2<T, T1, T2> = (
   currentValue: string | null,
-  arg1: T1,
-  arg2: T2,
+  arg1: any,
+  arg2: any,
```

Comment:
this breaks the automatic type resolution of the parameter types. This only affects users, which do not use the `@ReactNativeFlow` annotation, because with the annotation the correct signature is generated by the kotlin compiler which has access to the native type. This is a breaking change. But it is also not possible to know the type from the native signature in JS, because a string in the ts type could be a primitive string or a serialised object which cant be reconstructed in ts.

Reply:
True. Alternatively, we could leave the args properly typed and change the generation to also generate a wrapper for `@ReactNativeFlow` usages just like for `@ReactNativeMethod`:

```kotlin
@ReactNativeModule("MyRNModule")
class MyRnModule() {
  @ReactNativeFlow
  fun myFlow(myParam: List<String>): Flow<Int> = ...
}
``` 

generating to

```typescript
interface MyRNModuleInterface {
  myFlow: Next1<number, string[]>;
}

const MyRNModule: MyRNModuleInterface = {
  ...NativeModules.MyRNModule,
  myFlow: (currentValue: string | null, arg: string[]) => NativeModules.MyRNModule.myFlow(currentValue, arg),
}
```

`useFlow` could be left as it was before.
This would be the same behavior as not using `@ReactNativeFlow` but `.toReact` and  `@ReactNativeMethod`.

Reply:
That sounds better, can you implement it this way.

Reply:
Alright closing in favor of #31